src/qiskit_qec/circuits/css_code.py

```diff
@@ -0,0 +1,465 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=invalid-name, disable=no-name-in-module
+
+"""Generates circuits for CSS codes."""
+from qiskit import QuantumCircuit, QuantumRegister, ClassicalRegister
+from qiskit_aer.noise import depolarizing_error, pauli_error
+
+from stim import Circuit as StimCircuit
+from stim import target_rec as StimTarget_rec
+
+from qiskit_qec.utils import DecodingGraphNode
+from qiskit_qec.circuits.code_circuit import CodeCircuit
+from qiskit_qec.utils.stim_tools import (
+    noisify_circuit,
+    get_stim_circuits,
+    detector_error_model_to_rx_graph,
+)
+from qiskit_qec.codes import StabSubSystemCode
+from qiskit_qec.operators.pauli_list import PauliList
+from qiskit_qec.linear.symplectic import normalizer
+from qiskit_qec.exceptions import QiskitQECError
+
+
+class CssCodeCircuit(CodeCircuit):
+    """
+    CodeCircuit class for generic CSS codes.
+    """
+
+    def __init__(self, code, T, basis="z", round_schedule="zx", noise_model=None):
+        """
+        Args:
+            code: A CSS code
+            T: Number of syndrome measurement rounds
+            basis: basis for encoding ('x' or 'z')
+            round_schedule: Order in which to measureme gauge operators ('zx' or 'xz')
+            noise_model: Pauli noise model used in the construction of noisy circuits.
+            If a tuple, a pnenomological noise model is used with the entries being
+            probabity of depolarizing noise on code qubits between rounds and
+            probability of measurement errors, respectively.
+        """
+
+        super().__init__()
+
+        self.code = code
+        self._get_code_properties()
+        self.T = T
+        self.basis = basis
+        self.base = "0"
+        self.round_schedule = round_schedule
+        self.noise_model = noise_model
+        self._phenom = isinstance(noise_model, tuple)
+        if self._phenom:
+            p_depol, p_meas = self.noise_model
+            self._depol_error = depolarizing_error(p_depol, 1)
+            self._meas_error = pauli_error([("X", p_meas), ("I", 1 - p_meas)])
+
```

Comment:
You should split this next part(s) into smaller functions. If they are more general methods that can be useful outside this class put them into a new file for circuit method or utilities or something, otherwise keep them here.